src/libs/API.js

```diff
@@ -58,8 +75,6 @@ function addDefaultValuesToParameters(command, parameters) {
         // If we end up here with no authToken it means we are trying to make an API request before we are signed in.
         // In this case, we should cancel the current request by pausing the queue and clearing the remaining requests.
         if (!authToken) {
-            redirectToSignIn();
```

Comment:
Good question, it makes sense to remove that code too because that case is already handled with the `isReady` flag and we shouldn't block the request (pausing or clearing it) here if the logout is handled by the http request (407 `jsonCode` server response) when the `authToken` is expired or non-existent.

Also, I think we also should remove this code, no?

https://github.com/Expensify/App/blob/8ccf3b1a5ae1b0286b21e0bd95b4af80615bc936/src/libs/Network.js#L203-L208

I will remove all that code and test.

Reply:
Yeah I _think_ we can. My reasoning is something like:

1. We are now guaranteeing that both `authToken` and `credentials` are read from storage
2. If we are online and have no `authToken` but do have `credentials` then we should `reauthenticate`
3. If we are online and have no `authToken` or `credentials` then we should get signed out by `reauthenticate`

Reply:
The only thing that should "pause" the network request queue is the process of authenticating. But we should test this assumption to be sure it's correct.

Reply:
I tested without that code and everything seems to be working fine (I tested deleting and modifying the auth token and credentials from the browser local storage).

Reply:
>deleting and modifying the auth token

This should cause a reauthentication

>and credentials from the browser local storage

If we also do this then we should get logged out.

To clarify, is that what you mean by "working fine" or something else?

Reply:
Ah yeah, when I was deleting or modifying the auth token the reauthentication was triggered and also when I was wiping out the credentials + auth token the user was logged out